metagraph/src/seq_io/sequence_io.hpp

```diff
@@ -202,8 +203,33 @@ class FastaParser::iterator : public std::iterator<std::input_iterator_tag,
     }
 
   private:
-    iterator(const std::string &filename, bool with_reverse_complement);
-    void deinit_stream();
+    iterator(const std::string &filename, bool with_reverse_complement)
+          : filename_(filename),
+            with_reverse_complement_(with_reverse_complement) {
+        gzFile input_p = gzopen(filename_.c_str(), "r");
+        if (input_p == Z_NULL) {
+            std::cerr << "ERROR: Cannot read from file " << filename_ << std::endl;
+            exit(1);
+        }
+
+        read_stream_ = kseq_init(input_p);
+        if (read_stream_ == NULL) {
+            std::cerr << "ERROR: failed to initialize kseq file descriptor" << std::endl;
+            exit(1);
+        }
+
+        if (kseq_read(read_stream_) < 0)
+            deinit_stream();
+    }
+
+    void deinit_stream() {
+        if (read_stream_) {
+            gzFile input_p = read_stream_->f->f;
+            kseq_destroy(read_stream_);
+            gzclose(input_p);
+        }
+        read_stream_ = NULL;
+    }
```

Comment:
this was done to prevent gcc from complaining about kseq_init and kseq_destroy being unused in sequence_io.hpp

TODO: find a better way to do this

Reply:
would something like `(void) kseq_destroy;` solve the problem?

Reply:
Looking through `kseq.h`, I found that using `KSEQ_DECLARE` instead of `KSEQ_INIT` fixed the issue

Reply:
The issue now is that `sequence_io.cpp` has a sequence of macros taken from kseq.h to fill in missing declarations